docs/set_up_admin_tails.rst

```diff
@@ -55,25 +55,27 @@ Start by running the following commands to download the git repository.
 Verify the Release Tag
 ~~~~~~~~~~~~~~~~~~~~~~
 
-First, download the *Freedom of the Press Foundation Master Signing Key*
-and verify the fingerprint.
+First, download and verify the *Freedom of the Press Foundation Master
+Signing Key*. When passing the full public key fingerprint to the
+``--recv-key`` command, GPG will implicitly verify that the fingerprint
+of the key received matches the argument passed.
 
 .. code:: sh
 
-    gpg --keyserver pool.sks-keyservers.net --recv-key B89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818
-    gpg --fingerprint B89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818
+    gpg --keyserver hkp://qdigse2yzvuglcix.onion --recv-key B89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818
 
-The Freedom of the Press Foundation Master Signing Key should have a
-fingerprint of ``B89A 29DB 2128 160B 8E4B 1B4C BADD E0C7 FC9F
-6818``.
+It is important you type this out correctly. If you are not copy-pasting
+it, we recommend typing the fingerprint out with spaces like so ``B89A
+29DB 2128 160B 8E4B 1B4C BADD E0C7 FC9F 6818``, double-checking you have
+it correct, and then removing the spaces before pressing "enter."
 
-.. caution:: If the fingerprint does not match, fingerprint
-	     verification has failed and you **should not** proceed
-	     with the installation. If this happens, please contact us
-	     at securedrop@freedom.press.
+.. caution:: If GPG warns you that the fingerprint of the key received
+             does not match the one requested **do not** proceed with
+             the installation. If this happens, please email us at
+             securedrop@freedom.press.
```

Comment:
[Here's](http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blobdiff;f=g10/keyserver.c;h=dca5e18bf94c809938c5994d92179b67e4c4dbcf;hp=7bf9830645d32c9e3315ab4971a45b4c677cdb34;hb=5230304349490f31aa64ee2b69a8a2bc06bf7816;hpb=8eab483a1c4817a2946624c7305f464089d1875e) the original diff which added the `keyserver_retrieval_filter` function. [The patch](https://bugs.debian.org/cgi-bin/bugreport.cgi?att=1;bug=725411;msg=5;filename=0001-filter-and-verify-keyserver-responses.patch) Debian applied to the oldstable package seems to use this function in the original form (although it has since been rewritten). So when a bad key is given `keyserver_retrieval_filter` returns 1, which in turn causes the the `import_keys_stream` function on line 1502 of the 1.4.12 branch (unpatched) to fail. Not sure exactly what error message is propagated though.

The
```
/* TODO: Warn here if the fingerprint we got doesn't match the one
we asked for? */
```
warning is still present in the `keyserver_import_fprint` function (which starts on line 1675 on the unpatched 1.4.12 branch, is not changed by the patch). Seems like to you will get a generic error that does not specifically tell you the key you received has a fingerprint different than the one you request. What that generic error says exactly I'm unsure.

I feel like I've spent too much time on this as is.